### ISSUE TITLE
Allow disabling external diff drivers everywhere

### ISSUE THREAD

**Issue by piranha** [magit/magit#2069]
Hi! When I use git on a command line, I have configured external diff (namely `patdiff`). Unfortunately, I cannot configure its output to be similar enough to normal git's so that magit will recognize it.

So I decided to use magit with vanilla git diff and did this in my config:

```
(add-to-list 'magit-diff-arguments "--no-ext-diff")
(add-to-list 'magit-diff-section-arguments "--no-ext-diff")
```

And when I look at `magit-status`, everything works as it should. But when I press `d u` (for example), I see `patdiff` used. I tracked that down to function `magit-diff-read-args`, which calls `(magit-diff-arguments)` (not sure why is it called as a function, when it's a variable, maybe this misunderstanding is my problem?). And `(magit-diff-arguments)` returns nil there. 

Any ideas what's wrong here?

**Comment by tarsius** [comment]
To add a new argument you have to add it using `magit-define-popup-switch` or `...-option`.

`magit-diff-section-arguments` and the respective popup (`D`) is handled quite differently than all the other popups. A not sure if I should consider it a bug that just modifing the variable directly does appear to work (it probably stops working once you start using the `D` popup).

See `magit-popup`s manual, in particular http://magit.vc/manual/magit-popup/Customizing-existing-popups.html.

**Comment by piranha** [comment]
Thanks! Works beautifully.

**Comment by piranha** [comment]
Ok, so that worked for magit-diff-popup, but what do I do with commits? If I define `--no-ext-diff` for commit popup, `git commit` says `unknown option --no-ext-diff`, which is reasonable. But then when I edit commit message, emacs window with diff uses patdiff. :) Can I customize it's options as well?

**Comment by tarsius** [comment]
Implemented in #2119 (which will be merged soon).

**Comment by piranha** [comment]
Thanks a lot! I really appreciate development of magit. :+1:
